### PR TITLE
Warning log on new event during set workflow

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Warning log on new event during set workflow

**Why?**
Warning log on new event during set workflow

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/history/workflow/context.go [hunk 1]
```diff
@@ -680,7 +680,7 @@ func (c *ContextImpl) SetWorkflowExecution(ctx context.Context, now time.Time) (
 		return err
 	}
 	if len(resetWorkflowEventsSeq) != 0 {
-		return serviceerror.NewInternal("SetWorkflowExecution encountered new events")
+		c.logger.Warn("SetWorkflowExecution encountered new events")
```
Comment (yycptt) [inline service/history/workflow/context.go:683]: nit: also emit a metrics?

and we should revert the change before next release (1.19)

Comment (yux0) [reply service/history/workflow/context.go:684]: Added a metrics
